dask/array/core.py

```diff
@@ -3727,13 +3774,86 @@ def _check_regular_chunks(chunkset):
+    chunk in the dask array. This is the inverse of `to_delayed`.
 
-    The dask array will consist of a single chunk.
+    Notes
+    -----
+    If you're converting a dask Array to delayed with `to_delayed`, then later
+    converting it back with `from_delayed`, it's better to specify
+    ``.to_delayed(optimize_graph=False)``, so that array optimizations are applied just
+    once, at the end.
+
+    Parameters
+    ----------
+    value: ndarray or dask.delayed value
+        Single :class:`dask.delayed.Delayed` value, or a NumPy object array of them.
+        Each delayed value will become a chunk of the resulting dask array.
+    shape: tuple[int, ...], optional
+        The shape of the resulting dask array. One of ``shape``, ``chunks``, or ``like``
+        is required.
+    chunks: int, tuple; optional
+        The chunk shape of the resulting dask array. This should be the shape that each
+        delayed value will have, when computed. Can be given as:
+
+        * A blocksize like 1000.
+        * A blockshape like (1000, 1000).
+        * Explicit sizes of all blocks along all dimensions like ((1000, 1000, 500),
+          (400, 400)).
+
+        One of ``shape``, ``chunks``, or ``like`` is required.
+    dtype: NumPy dtype, optional
+        The dtype of the dask array. This should be the dtype that every delayed value
+        will have when computed. One of ``dtype``, ``meta``, or ``like`` is required.
+    meta: array-like, optional
+        The ``meta`` of the output array. If given, should be an array-like of the same
+        type and dtype that would result from calling ``.compute()`` on every delayed
+        value. One of ``dtype``, ``meta``, or ``like`` is required.
+    name: str, optional
+        The key name to use for the output array. Note that this fully specifies the
+        output key name, and must be unique. If not provided, will be determined by a
+        hash of the arguments.
+    like: dask Array, optional
+        Dask array to use as reference. The ``chunks`` and ``meta`` parameters will be
+        copied from this example if they are not given.
+    allow_unknown_chunksizes: bool, default False
+        If False, and ``shape``, ``chunks``, or ``like`` are not given, a `ValueError`
+        is raised, since the size of each chunk would be unknown.
+
+        If it's impossible to know the size that each delayed value will be when
+        computed (for example, if that depends on data that's loaded), you can set
+        ``allow_unknown_chunksizes=True`` to make the chunksize all NaNs. This is
+        generally not recommended, because some operations (like slicing) require known
+        chunksizes.
+
+        Note that even with unknown chunksizes, all chunks must have the same number of
+        dimensions. Furthermore, along every "row" or "column" of chunks, each chunk
+        must be the same length in that dimension---jagged arrays are not supported and
+        can cause unexpected results.
```

Comment:
Thoughts about removing this an instead raising a warning where appropriate, pointing users towards `compute_chunk_sizes()`? I could see it going either way.

Reply:
I definitely thought about doing a warning instead. I don't have a preference either way. I just happened to notice the `allow_unknown_chunksizes` parameter in other places (like `concatenate`) so I figured I'd follow that pattern. Mostly I thought it might be too easy to create unknown chunksizes out of laziness (not passing a shape/chunksize) and not reading a warning.

Reply:
No strong feelings here, I'm happy to have this kwarg